testutil/src/main/java/org/spine3/test/NullToleranceTest.java

```diff
@@ -0,0 +1,364 @@
+import static com.google.common.collect.Sets.newHashSet;
+import static com.google.common.primitives.Primitives.allPrimitiveTypes;
+
+/**
+ * Serves as a helper to ensure that none of the methods of the target utility
+ * class accept {@code null}s as argument values.
+ *
+ * <p> The helper checks the methods with access modifiers:
+ * <ul>
+ *     <li> the {@code public};
+ *     <li> the {@code protected};
+ *     <li> the {@code default}.
+ * </ul>
+ *
+ * <p> The helper does not checks the methods:
+ * <ul>
+ *     <li> with the {@code private} modifier;
+ *     <li> without the {@code static} modifier;
+ *     <li> with only the primitive parameters.
+ * </ul>
+ *
+ * <p> The examples of the methods which will be checked:
+ * <ul>
+ *     <li> public static void method(Object obj);
+ *     <li> protected static void method(Object first, long second);
+ *     <li> static void method(Object first, Object second).
+ * </ul>
+ *
+ * <p> The examples of the methods which will be ignored:
+ * <ul>
+ *     <li> public void method(Object obj);
+ *     <li> private static void method(Object obj);
+ *     <li> protected static void method(int first, float second).
+ * </ul>
+ *
+ * @author Illia Shepilov
+ */
+public class NullToleranceTest {
+
+    private final Class targetClass;
+    private final Set<String> excludedMethods;
+    private final Map<?, ?> defaultValuesMap;
+
+    private NullToleranceTest(Builder builder) {
+        this.targetClass = builder.targetClass;
+        this.excludedMethods = builder.excludedMethods;
+        this.defaultValuesMap = builder.defaultValues;
+    }
+
+    /**
+     * Checks the all non-private methods in the {@code targetClass}.
+     *
+     * <p> Check is successful if each of the non-primitive method parameters is ensured to be non-null.
+     *
+     * @return {@code true} if all methods have not-null check
+     * for the input reference type parameters, {@code false} otherwise
+     */
+    public boolean check() {
+        final Method[] accessibleMethods = getAccessibleMethods();
+        for (Method method : accessibleMethods) {
+            final Class[] parameterTypes = method.getParameterTypes();
+            final String methodName = method.getName();
+            final boolean excluded = excludedMethods.contains(methodName);
+            final boolean arePrimitives = allPrimitiveTypes().containsAll(Arrays.asList(parameterTypes));
+            final boolean skipIteration = excluded || parameterTypes.length == 0 || arePrimitives;
+            if (skipIteration) {
+                continue;
+            }
+
+            final Object[] parameterValues = getParameterValues(parameterTypes);
+
+            final boolean correct = invokeAndCheck(method, parameterValues, parameterTypes);
+            if (!correct) {
+                return false;
+            }
+
+        }
+        return true;
+    }
+
+    private Method[] getAccessibleMethods() {
```

Comment:
Some of `NullToleranceTest ` methods (and this one is a good example) can be grouped into an inner utility class.